src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -45,39 +46,54 @@ public KeysController(WebScriptHostManager scriptHostManager, ISecretManager sec
         {
-            var hostSecrets = await _secretManager.GetHostSecretsAsync();
-            return GetKeysResult(hostSecrets.FunctionKeys);
+            string hostKeyScope = GetHostKeyScopeForRequest();
+
+            Dictionary<string, string> keys = await GetHostSecretsByScope(hostKeyScope);
+            return GetKeysResult(keys);
         }
 
         [HttpPost]
         [Route("admin/functions/{name}/keys/{keyName}")]
-        public Task<IHttpActionResult> Post(string name, string keyName) => AddOrUpdateFunctionSecretAsync(keyName, null, name);
+        public Task<IHttpActionResult> Post(string name, string keyName) => AddOrUpdateSecretAsync(keyName, null, name, ScriptSecretsType.Function);
 
         [HttpPost]
-        [Route("admin/host/keys/{keyName}")]
-        public Task<IHttpActionResult> Post(string keyName) => AddOrUpdateFunctionSecretAsync(keyName, null);
+        [Route("admin/host/{keys:regex(^(keys|functionkeys|systemkeys)$)}/{keyName}")]
+        public Task<IHttpActionResult> Post(string keyName) => AddOrUpdateSecretAsync(keyName, null, GetHostKeyScopeForRequest(), ScriptSecretsType.Host);
 
         [HttpPut]
         [Route("admin/functions/{name}/keys/{keyName}")]
-        public Task<IHttpActionResult> Put(string name, string keyName, Key key) => PutKeyAsync(keyName, key, name);
+        public Task<IHttpActionResult> Put(string name, string keyName, Key key) => PutKeyAsync(keyName, key, name, ScriptSecretsType.Function);
 
         [HttpPut]
-        [Route("admin/host/keys/{keyName}")]
-        public Task<IHttpActionResult> Put(string keyName, Key key) => PutKeyAsync(keyName, key);
+        [Route("admin/host/{keys:regex(^(keys|functionkeys|systemkeys)$)}/{keyName}")]
+        public Task<IHttpActionResult> Put(string keyName, Key key) => PutKeyAsync(keyName, key, GetHostKeyScopeForRequest(), ScriptSecretsType.Host);
 
         [HttpDelete]
         [Route("admin/functions/{name}/keys/{keyName}")]
-        public Task<IHttpActionResult> Delete(string name, string keyName) => DeleteFunctionSecretAsync(keyName, name);
+        public Task<IHttpActionResult> Delete(string name, string keyName) => DeleteFunctionSecretAsync(keyName, name, ScriptSecretsType.Function);
 
         [HttpDelete]
-        [Route("admin/host/keys/{keyName}")]
-        public Task<IHttpActionResult> Delete(string keyName) => DeleteFunctionSecretAsync(keyName);
+        [Route("admin/host/{keys:regex(^(keys|functionkeys|systemkeys)$)}/{keyName}")]
+        public Task<IHttpActionResult> Delete(string keyName) => DeleteFunctionSecretAsync(keyName, GetHostKeyScopeForRequest(), ScriptSecretsType.Host);
+
+        private string GetHostKeyScopeForRequest()
+        {
+            string keyScope = ControllerContext.RouteData.Values.GetValueOrDefault<string>("keys");
+
+            if (string.Equals(keyScope, "keys"))
+            {
+                keyScope = HostKeyScopes.FunctionKeys;
+            }
+
+            return keyScope;
```

Comment:
You have an enum like Type HostKeyScopes - wonder if it shouldn't be an actual enum that you flow around rather than a string / string compares?